File: April-Challenges-2021/Day-30/Powerful-Integers.cpp
class Solution {
public:
    vector<int> powerfulIntegers(int x, int y, int bound) 
    {
        set<int> powerfull;
        
        for (size_t i = 1; i <= bound; i *= x)
        {
            for (size_t j = 1; i + j <= bound; j *= y)
            {
                powerfull.insert(i + j);
                
                if (y == 1)
                    break;
            }
            if (x == 1)
                break;
        }
        
        return vector(powerfull.begin(), powerfull.end());
    }
};

File: April-Challenges-2021/Day-30/Powerful-Integers.py
class Solution:
    def powerfulIntegers(self, x: int, y: int, bound: int) -> List[int]:
        p = []
        for i in range(20):
            for j in range(20):
                t = x**i+y**j
                if t <= bound and t not in p:
                    p.append(t)
        return sorted(p)
